### PR TITLE
lite2/rpc: verify block results and validators

### PR DESCRIPTION
**Pull request by melekes**
Closes: #4695

## Description

Verify `/block_results` and `/validators` responses from an HTTP client using the light client.

Added `count` and `total` to `/validators` response.

Refs #3113
______

For contributor use:

- [ ] Wrote tests
- [ ] Updated CHANGELOG_PENDING.md
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: lite2/rpc/client.go [hunk 1]
```diff
@@ -340,7 +403,7 @@ func (c *Client) UnsubscribeAll(ctx context.Context, subscriber string) error {
 
 func (c *Client) updateLiteClientIfNeededTo(height int64) (*types.SignedHeader, error) {
 	h, err := c.lc.VerifyHeaderAtHeight(height, time.Now())
-	return h, errors.Wrapf(err, "failed to update light client to %d", height)
+	return h, fmt.Errorf("failed to update light client to %d: %w", height, err)
```
Comment (cmwaters) [inline lite2/rpc/client.go:406]: Why does this always return an error?

File: lite2/rpc/client.go [hunk 2]
```diff
@@ -317,8 +352,36 @@ func (c *Client) TxSearch(query string, prove bool, page, perPage int, orderBy s
...
+		return nil, err
+	}
+
+	// Validate res.
+	if res.BlockHeight <= 0 {
+		return nil, errNegOrZeroHeight
+	}
+
+	// Update the light client if we're behind.
+	h, err := c.updateLiteClientIfNeededTo(res.BlockHeight)
```
Comment (cmwaters) [inline lite2/rpc/client.go:371]: do you think this is the right name for the function. Aren't we just getting a verified header to compare the validator set with

Comment (melekes) [reply lite2/rpc/client.go:371]: but it calls VerifyHeaderAtHeight

Comment (cmwaters) [reply lite2/rpc/client.go:371]: Yeah I'm not too fussed - I just thought that the function name wasn't as helpful

**Pull request merged**